pkg/appgw/internaltypes.go

```diff
@@ -116,9 +117,9 @@ func (s secretIdentifier) secretKey() string {
 func (s secretIdentifier) secretFullName() string {
 	// in case of referring ssl certificate from agic annotation
 	if len(s.Namespace) == 0 {
-		return s.Name
+		return fmt.Sprintf("%v%v-%v", agPrefix, prefixSslCertificate, s.Name)
```

Comment:
One use case about null namespace here, i.e. [appgw-ssl-certificate](https://github.com/Azure/application-gateway-kubernetes-ingress/blob/master/docs/annotations.md#appgw-ssl-certificate), is that the secret will be pre-installed on AppGw with a user-specified name. AppGw will not be able to reference the right sslCertificate resource if the prefix added here.

Reply:
Hi @3quanfeng ,
You are right! I understood bad the way as agic works but your commentary about it is very logic. I will change it during the day

Reply:
Done @3quanfeng ! :)